### PR TITLE
Add minor text clarifications to Data Migrations docs

### PR DESCRIPTION
**Pull request by brutoledo**
Hi guys,
I'm a newcomer to OrchardCore and Im studying the project. 

As a disclaimer, it's my very first time contributing to an open-source project and I just want to improve a bit the Data Migrations document page :)

Context:
Reading the current doc explanation about a module with multiple features, gave me the idea that is also possible to have multiple migrations (class files) per feature when in reality it's only possible to have one migration file per feature.

I'm also adding a first-level heading to make it easier to scroll up the page using a table of contents.

...

I'm sorry if it's a stupid PR but I spent some time trying to put it into practice until I realize it's not actually supported/ how it should be. Therefore I just would like to make the text a bit better.

### CODE REVIEW

**Comment by brutoledo** [comment]
@dotnet-policy-service agree

File: src/docs/reference/modules/Migrations/README.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # Data Migrations (`OrchardCore.Data.Migration.DataMigration`)
 
+## Configuring Data Migration
+
 Data Migration classes can be used to alter the content type definitions (like by adding new __types__, or configuring their __parts__ and __fields__), 
 initializing recipes or creating indices.
 
-There should be one Migrations file per module inheriting from __DataMigration__. However, if a module has more than one feature and multiple migrations per feature, each migration should be decorated with the Feature attribute i.e. `[Feature("OrchardCore.ContentFields.Indexing.SQL")]`
+There should be one Migrations file per module inheriting from __DataMigration__. However, if a module has more than one feature, it can be added one migration file per feature, being decorated with the Feature attribute i.e. `[Feature("OrchardCore.ContentFields.Indexing.SQL")]`
```
Comment (MikeAlhayek) [inline src/docs/reference/modules/Migrations/README.md:8]: The docs seems to be outdated event with the suggested test.  Please remove this paragraph altogether.

File: src/docs/reference/modules/Migrations/README.md [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
 Initial migration method should be named `public int Create` or `public Task<int> CreateAsync` and it should return a number (like  `return 1`).
```
Comment (MikeAlhayek) [inline src/docs/reference/modules/Migrations/README.md:10]: Also, please replace this paragraph with the following 

```
To implement a migration, your class should inherit from the DataMigration class. The initial migration method must be named `public int Create()` or `public Task<int> CreateAsync()`, and it should return a number (e.g., 1). Subsequent migrations should adhere to the convention public int `UpdateFromX()` or `public Task<int> UpdateFromXAsync()`, where `X` represents the number returned by the last migration method. The migration process strictly follows this chain: for instance, `UpdateFrom4()` will only execute if the last migration returned the number `4`. It is crucial not to modify these numbers afterward, as doing so may disrupt migrations for other tenants.
```

**Comment by MikeAlhayek** [comment]
Merging...

@brutoledo Thanks for contributing to OrchardCore! With your help contributing will make the project better. I encourage you to continue on contributing.

**Pull request merged**